sdk/identity/src/token_credentials/client_certificate_credentials.rs

```diff
@@ -130,16 +135,17 @@ struct AadTokenResponse {
     access_token: String,
 }
 
-fn get_encoded_cert(cert: &X509) -> Result<String, ClientCertificateCredentialError> {
+fn get_encoded_cert(cert: &X509) -> azure_core::Result<String> {
     Ok(format!(
         "\"{}\"",
-        base64::encode(
-            cert.to_pem()
-                .map_err(ClientCertificateCredentialError::OpensslError)?
-        )
+        base64::encode(cert.to_pem().map_err(|_| openssl_error())?)
     ))
 }
 
+fn openssl_error() -> azure_core::error::Error {
+    Error::message(ErrorKind::Credential, "Openssl decode error")
+}
```

Comment:
I don't think we want to use the same error message for all of these errors. It makes it hard to tell what actually went wrong.